src/analyzing/AST_to_IL.ml

```diff
@@ -1027,16 +1027,32 @@ and record env ((_tok, origfields, _) as record_def) =
                    ( { G.name = G.EN (G.Id (id, _)); tparams = None; _ },
                      def_kind );
                _;
-             } ->
-             let fdeforig =
+             } as forig ->
+             let field_def =
                match def_kind with
                (* TODO: Consider what to do with vtype. *)
                | G.VarDef { G.vinit = Some fdeforig; _ }
                | G.FieldDefColon { G.vinit = Some fdeforig; _ } ->
-                   fdeforig
+                   expr env fdeforig
+               | G.FuncDef fdef ->
```

Comment:
Could you add a comment indicating where this case may be coming from? LIke `(* e.g. a lambda in JS *)` or something like that.